packages/serverpod_serialization/lib/src/serialization.dart

```diff
@@ -253,8 +253,30 @@ abstract class SerializationManager {
   /// name so that it can be decoded even if th class is unknown.
   /// If [formatted] is true, the output will be formatted with two spaces
   /// indentation.
-  String encodeWithType(Object? object, {bool formatted = false}) {
-    return encode(wrapWithClassName(object), formatted: formatted);
+  String encodeWithType(
+    Object? object, {
+    bool formatted = false,
+  }) {
+    return encode(
+      wrapWithClassName(object),
+      formatted: formatted,
+    );
+  }
+
+  /// Encode the provided [object] to a Json-formatted [String], including the
+  /// class name so that it can be decoded even if the class is unknown.
+  /// If [formatted] is true, the output will be formatted with two spaces
+  /// indentation. If [object] implements [ProtocolSerialization] interface, then
+  /// [toJsonForProtocol] will be used instead of the [toJson] method.
+  String encodeForProtocolWithType(
```

Comment:
Suggestion: This ensures that we find both methods when we start in the same way. (same as toJson and toJsonForProtocol).

```suggestion
  String encodeWithTypeForProtocol(
```

Reply:
If you do this renaming, remember to change the name in all the test as well.